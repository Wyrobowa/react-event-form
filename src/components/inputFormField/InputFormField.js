import React from 'react';
import PropTypes from 'prop-types';

const InputFormField = ({
  id, label, value, type, onChange,
}) => (
  <div className="form-group">
    <label htmlFor={id}>{label}</label>
    <input
      className="form-control"
      id={id}
      name={id}
      value={value}
      type={type}
      onChange={onChange}
    />
  </div>
);

InputFormField.propTypes = {
  id: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  value: PropTypes.oneOfType([
    PropTypes.number,
    PropTypes.string,
  ]),
  type: PropTypes.oneOf(['text', 'email', 'date']),
  onChange: PropTypes.func.isRequired,
};

InputFormField.defaultProps = {
  value: '',
  type: 'text',
};

export default InputFormField;
