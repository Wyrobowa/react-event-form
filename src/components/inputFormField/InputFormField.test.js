import React from 'react';
import { shallow } from 'enzyme';
import InputFormField from './InputFormField';

describe('Header', () => {
  it('renders InputFormField component without crashing', () => {
    const component = shallow(<InputFormField id="name" label="Name" onChange={() => {}} />);

    expect(component).toMatchSnapshot();
  });
});
