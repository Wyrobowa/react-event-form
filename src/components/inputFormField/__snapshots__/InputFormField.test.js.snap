// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders InputFormField component without crashing 1`] = `
<div
  className="form-group"
>
  <label
    htmlFor="name"
  >
    Name
  </label>
  <input
    className="form-control"
    id="name"
    name="name"
    onChange={[Function]}
    type="text"
    value=""
  />
</div>
`;
